### PR TITLE
Use version label in app and appcatalog controllers

### PR DESCRIPTION
**Pull request by rossf7**
Towards https://github.com/giantswarm/giantswarm/issues/4845

### CODE REVIEW

File: pkg/label/label.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
+// Package label contains common Kubernetes object labels. These are defined in
+// https://github.com/giantswarm/fmt/blob/master/kubernetes/annotations_and_labels.md.
+package label
```
Comment (rossf7) [inline pkg/label/label.go]: Adding a label package like we have in cluster-operator. As app-operator has 2 controllers. So we need shared code.

File: service/controller/app/v1/key/key.go [hunk 2]
```diff
@@ -77,8 +75,10 @@ func Version(customResource v1alpha1.App) string {
 	return customResource.Spec.Version
 }
 
-func VersionBundleVersion(customResource v1alpha1.App) string {
-	if val, ok := customResource.ObjectMeta.Annotations[versionBundleAnnotation]; ok {
+// VersionLabel returns the label value to determine if the custom resource is
+// supported by this version of the operatorkit resource.
+func VersionLabel(customResource v1alpha1.App) string {
+	if val, ok := customResource.ObjectMeta.Labels[label.Version]; ok {
```
Comment (rossf7) [inline service/controller/app/v1/key/key.go]: Key helper for label replaces previous annotation.

File: service/controller/appcatalog/v1/key/key.go [hunk 3]
```diff
@@ -32,8 +33,10 @@ func ToCustomResource(v interface{}) (v1alpha1.AppCatalog, error) {
 	return *customResource, nil
 }
 
-func VersionBundleVersion(customObject v1alpha1.AppCatalog) string {
-	if val, ok := customObject.ObjectMeta.Annotations[versionBundleAnnotation]; ok {
+// VersionLabel returns the label value to determine if the custom resource is
+// supported by this version of the operatorkit resource.
+func VersionLabel(customResource v1alpha1.AppCatalog) string {
+	if val, ok := customResource.ObjectMeta.Labels[label.Version]; ok {
```
Comment (rossf7) [inline service/controller/appcatalog/v1/key/key.go]: Same change for appcatalog controller.

File: pkg/label/label.go [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+const (
+	// ChartOperatorVersion is set for chart CRs managed by the operator.
+	ChartOperatorVersion = "chart-operator.giantswarm.io/version"
+
+	// ManagedBy is set for Kubernetes resources managed by the operator.
+	ManagedBy = "giantswarm.io/managed-by"
+
+	// Version is used to determine if the custom resource is supported by this
+	// version of the operatorkit resource.
+	Version = "app-operator.giantswarm.io/version"
```
Comment (xh3b4sd) [inline pkg/label/label.go]:
```suggestion
	AppOperatorVersion = "app-operator.giantswarm.io/version"
```

File: service/controller/app/v1/key/key.go (hunk 2 above)
Comment (xh3b4sd) [inline service/controller/app/v1/key/key.go]:
```suggestion
	if val, ok := customResource.ObjectMeta.Labels[label.AppOperatorVersion]; ok {
```

File: service/controller/app/v1/key/key_test.go [hunk 5]
```diff
@@ -237,43 +237,59 @@ func Test_Version(t *testing.T) {
 	}
 }
 
-func TestVersionBundleVersion(t *testing.T) {
+func Test_VersionLabel(t *testing.T) {
```
Comment (xh3b4sd) [inline service/controller/app/v1/key/key_test.go]:
```suggestion
func Test_Label_AppOperatorVersion(t *testing.T) {
```

File: service/controller/appcatalog/v1/key/key.go (hunk 3 above)
Comment (xh3b4sd) [inline service/controller/appcatalog/v1/key/key.go]:
```suggestion
	if val, ok := customResource.ObjectMeta.Labels[label.AppOperatorVersion]; ok {
```

File: service/controller/appcatalog/v1/key/key_test.go [hunk 6]
```diff
@@ -102,45 +102,48 @@ func Test_ToCustomResource(t *testing.T) {
 	}
 }
 
-func TestVersionBundleVersion(t *testing.T) {
+func Test_VersionLabel(t *testing.T) {
```
Comment (xh3b4sd) [inline service/controller/appcatalog/v1/key/key_test.go]:
```suggestion
func Test_Label_AppOperatorVersion(t *testing.T) {
```

File: service/controller/appcatalog/v1/key/key.go [hunk 7]
```diff
@@ -2,6 +2,7 @@ package key
 
 import (
 	"github.com/giantswarm/apiextensions/pkg/apis/application/v1alpha1"
+	"github.com/giantswarm/app-operator/pkg/label"
```
[the remark this answers is not in the document]
Comment (xh3b4sd) [reply service/controller/appcatalog/v1/key/key.go]: Yep, he got you. :D

[the remark this answers is not in the document]
Comment (rossf7) [reply service/controller/appcatalog/v1/key/key.go]: You sure did. 😆 ❤️

**Comment by rossf7** [comment]
@xh3b4sd @tomahawk28 Thx for the reviews. Good to make the version label naming explicit and I'll get those imports fixed. 😄
 
I'll get this sorted after my 1:1 with Ferran.

**Pull request merged**